### ISSUE TITLE
i18n files

### ISSUE THREAD

**Issue by grconrad** [github/linguist#5492]
## Describe the enhancement

We'd like to see a count of our localized text files on the repository's language breakdown. Is there a way to achieve this?

Our automated processes store translated strings in *.properties files in our project (e.g. i18n/checkout_de_DE.properties).

Because of the definitions and file extension mappings in languages.yml, these get classified as INI files; but `type: data` means they are not shown on the repository's languages breakdown.

Using .gitattributes we can map the file paths to languages defined in languages.yml, but no language seems suitable to describe translated text. And AFAICT there is currently no way for us to define a custom "I18n" entry in order to map file paths to it (via .gitattributes) so that those counts can be displayed on the repository's main page.

**Comment by lildude** [comment]
> We'd like to see a count of our localized text files on the repository's language breakdown. Is there a way to achieve this?

No, not to explicitly identify them as "i8n" or similar. The closest you can get is to identify another language (with the risk syntax highlighting will be wrong) and use an override or allow Linguist to determine if the files are Java Properties or INI files (the two languages the `.properties` ext is  currently associated with) as it does now.

> these get classified as INI files; but `type: data` means they are not shown on the repository's languages breakdown.
> 
> Using .gitattributes we can map the file paths to languages defined in languages.yml, but no language seems suitable to describe translated text. And AFAICT there is currently no way for us to define a custom "I18n" entry in order to map file paths to it (via .gitattributes) so that those counts can be displayed on the repository's main page.

If you're only wanting something to appear in the sidebar that maps to these files, you can implement the `linguist-detectable` override to force these to show in the sidebar, but they will be identified as Java Properties or INI, depending on the content.

If there's a way to uniquely identify them by content (so the heuristics can differentiate them from Java Properties or INI files), or explicit filenames (Linguist doesn't support using globs or patterns in explicit filenames) and there's enough usage, a new "i8n" entry could be added, however it would also need to be classified as "data" and thus hidden by default too. From a quick look of those files I can find on GitHub, they don't appear to be any different to Java Properties or INI files so I'm not sure this is going to be possible.

**Comment by grconrad** [comment]
Thank you @lildude .

Displaying our localization files as Java properties or INI is likely to be confusing for users.

Since linguist is programming-language-centric and only identifies language by file extension (whereas i18n is a fuzzier designation that doesn't neatly or uniquely map to a language), I agree with your assessment that there's no clean way to do what I want with linguist. It's probably not the right tool.

Properties files and INI files can be used for multiple purposes, and localizable text is one such purpose, but there are other purposes including generic configuration. I don't think there would be any content-based way to reliably discern the purpose of these files, as they are basically just key/value pairs with no metadata.

With path glob support perhaps consumers could configure that everything in, say, `/path/to/my/i18n/**/*.myextension` should be counted as i18n, but that implementation would be significantly different to how linguist works today.

**Comment by Nixinova** [comment]
There is [`extension:lang`](https://github.com/search?q=extension%3Alang&type=Code) with 1.2M results, which is mostly JSONC & `key=val` formats. Not sure what the best language name for this extension would be.

**Comment by gabrielcsapo** [comment]
@Nixinova having .lang would be great... this literally looks like our properties files with a different extension name. 

I think this might be a result of what https://www.rws.com/translation/language-cloud/management-system/worldserver/ generates and consumes. This is the industry standard for internationalization from what I can tell. (e.g at eBay, Paypal and LinkedIn they use that so my context might be skewed)
